keras_cv/layers/preprocessing/resizing.py

```diff
@@ -255,24 +308,58 @@ def _resize_with_crop(self, inputs):
         else:
             input_dtype = tf.float32
 
-        def resize_with_crop_to_aspect(x):
+        def resize_with_crop_to_aspect_images(x, interpolation_method):
             if isinstance(x, tf.RaggedTensor):
                 x = x.to_tensor()
             return keras.preprocessing.image.smart_resize(
-                x, size=size, interpolation=self._interpolation_method
+                x,
+                size=size,
+                interpolation=interpolation_method,
             )
 
         if isinstance(images, tf.RaggedTensor):
             size_as_shape = tf.TensorShape(size)
             shape = size_as_shape + images.shape[-1:]
             spec = tf.TensorSpec(shape, input_dtype)
+            map_fn = partial(
```

Comment:
Ahh I didn't realize we'd end up having to use a `partial` here -- my bad.

How about we keep a base method `resize_with_crop_to_aspect` and make two wrapper methods with the same name and `_images` and `_masks` suffixes which pass through the interpolation method (just to avoid using a partial)